xarray/tests/test_groupby.py

```diff
@@ -724,6 +728,12 @@ def test_groupby_getitem(dataset) -> None:
     )
     assert_identical(dataset.foo.sel(z=[1]), dataset.foo.groupby("z", squeeze=False)[1])
 
+    assert_identical(dataset.cat.sel(y=[1]), dataset.cat.groupby("y", squeeze=False)[1])
+    with pytest.raises(
```

Comment:
Can you open an issue for some groupby ops that you'd like to see work please?

Reply:
Well groupby is supported, but not when the groupby op would force a broadcast over i.e., 2 indices of an extension array.  That's why the above groupby over a `y` op works - the categorical is aligned with that index